airbyte-api/build.gradle

```diff
@@ -36,7 +36,8 @@ task generateApiServer(type: GenerateTask) {
     configOptions = [
             dateLibrary  : "java8",
             generatePom  : "false",
-            interfaceOnly: "true"
+            interfaceOnly: "true",
+            additionalModelTypeAnnotations: "\n@com.fasterxml.jackson.annotation.JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL)"
```

Comment:
I'm not sure if it is possible with these annotations, but it seems like the ideal behavior would be that fields with `nullable: true` are included in the serialized JSON if they are explicitly set to something (either null or a real value), and they are not included if they are not set. But fields that do _not_ have `nullable: true` should only ever be included in the serialized JSON if they are set to a non-null value, and they should not be included in the serialized JSON if they are set to null.

This behavior would always keep the openAPI spec consistent with the behavior of the API, while still allowing us to explicitly set fields to null when needed (if they are nullable)

Reply:
derp. agreed with what lake said. if we need it to be nullable then we should set it as such. okay. i think the implementation in this PR make sense.

Reply:
Thank you @cgardens for the example! They are correct.
> I think we want this:
```
// object in memory => serialized object
{ firstName: "charles" } => { firstName: "charles" }
{ firstName: "charles", lastName: null } => { firstName: "charles", lastName: null }
```


This is ideally what we want but I don't think it's achievable. When we create a model instance (from generated classes) the model attribute default to `null` in JAX-RS. It means that if we don't set the attribute on initialization the model instance will still have a `null` value for non-set attributes. So the default behavior will be:

```
// object in memory -> model instance in memory -> serialized json
{ firstName: "charles" } => { firstName: "Charles", lastName: "null" } =>  { firstName: "Charles", lastName: "null" } 
{ firstName: "charles", lastName: "null" } => { firstName: "Charles", lastName: "null" } =>  { firstName: "Charles", lastName: "null" } 
```

The change I made with the annotation leads to the following:
```
// object in memory -> model instance in memory -> serialized json
{ firstName: "charles" } => { firstName: "Charles", lastName: "null" } =>  { firstName: "Charles" } 
{ firstName: "charles", lastName: "null" } => { firstName: "Charles", lastName: "null" } =>  { firstName: "Charles"} 
```
Jackson cleans the object but without extra logic coming from the OpenAPI spec. **It means that even if we set `nullable = true` on the `lastName` attribute we won't end up with a `null` value in the payload.**
It answers @lmossman's question:
> I'm not sure if it is possible with these annotations, but it seems like the ideal behavior would be that fields with nullable: true are included in the serialized JSON if they are explicitly set to something.

This is not possible with this implementation.

@cgardens @lmossman @jdpgrailsdev I would like to know if you are ok with this and want you to be aware that with this change the `nullable = true` won't be of any effect in the Open API spec. I am under the impression it is an acceptable compromise at the moment as we are not using this property but we might want to find a more flexible solution in the future. I don't know what this solution could be but it might be changing the generator from `jaxrs-spec` to another like `spring` 🤔

Reply:
Got it, it sounds like we can't get the ideal behavior we want with the current libraries we are using then. In that case, especially since it seems we aren't currently using nullable fields, I think your current implementation is acceptable. 

@alafanechere I think it would be helpful to add a comment for this line that summarizes what we discussed here, to help explain why we are setting it to `NOT_NULL` and what that means for nullable fields.